datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -2206,6 +2237,14 @@ private void populateDatastreamDestinationFromExistingDatastream(Datastream data
             existingStream.getMetadata().get(DatastreamMetadataConstants.TASK_PREFIX));
   }
 
+  // Via the intrinsic conditional variable, notify other threads that might
+  // be waiting on acquiring access on the Coordinator object.
+  // We are only calling notify on the synchronized Coordinator Object's ("this") waiting threads.
+  // Suppressing the Naked_Notify warning on this.
+  protected synchronized void notifyThreadsWaitingForCoordinatorObjectSynchronization() {
```

Comment:
do we need this helper method?

Reply:
I wanted to use this from couple of tests and hence decided to keep it in a generic function which can be reused.